### PR TITLE
fix(next): thread default ServerProps to view actions and other components that were missing

### PR DESCRIPTION
**Pull request by akhrarovsaid**
### What?
This PR threads default `serverProps` to Edit & List view action slots.

### Why?
To be more in-line with how the docs specify custom components should work. Custom components should recieve, at minimum, the `BasePayload` instance and `i18n` object.

### How?
By including props into `RenderServerComponent` call in the `DefaultTemplate`

Notes:
- There is an opportunity for a refactor here to reuse these props throughout multiple places in `DefaultTemplate`. Slightly awkward due to having the actions being memoized but doable. Happy to make the necessary changes for this.

### CODE REVIEW

File: packages/next/src/templates/Default/index.tsx [hunk 1]
```diff
@@ -51,6 +51,15 @@ export const DefaultTemplate: React.FC<DefaultTemplateProps> = ({
   const { Actions } = React.useMemo<{
     Actions: Record<string, React.ReactNode>
   }>(() => {
+    const viewActionServerProps = {
```
Comment (jacobsfletch) [inline packages/next/src/templates/Default/index.tsx:54]: Rename this to `serverProps` and type is as `ServerProps` (imported from `payload`). Can these also be defined outside the `useMemo`?

Comment (akhrarovsaid) [reply packages/next/src/templates/Default/index.tsx:54]: They can but they still have to be memoized. I can also pass these "shared" server props to the other components defined in the same file.

Comment (jacobsfletch) [reply packages/next/src/templates/Default/index.tsx:54]: If that's an issue we can memoize the `serverProps` object itself, and then pass that around to the other components in this layout like you've suggested.

Comment (akhrarovsaid) [reply packages/next/src/templates/Default/index.tsx:54]: @jacobsfletch I also realized that the `avatar` and `graphics.Icon` here are not recieving serverProps - shall I include them?

Comment (akhrarovsaid) [reply packages/next/src/templates/Default/index.tsx]: Moved and memoized serverProps, added type as discussed.

File: packages/next/src/templates/Default/index.tsx [hunk 2]
```diff
@@ -72,21 +87,15 @@ export const DefaultTemplate: React.FC<DefaultTemplateProps> = ({
...
     importMap: payload.importMap,
     serverProps: {
-      i18n,
-      locale,
-      params,
-      payload,
-      permissions,
-      searchParams,
-      user,
+      ...serverProps,
```
Comment (jacobsfletch) [inline packages/next/src/templates/Default/index.tsx:98]: We shouldn't spread here, as it can be slow.

**Review by jacobsfletch — APPROVED** [review]

**Comment by akhrarovsaid** [comment]
Thanks @jacobsfletch & @AlessioGr for the review and additions! Appreciate it!

**Comment by jacobsfletch** [comment]
@AlessioGr is going to add include a few more components to this which were also missing server props.

**Review by AlessioGr — APPROVED** [review]

**Pull request merged**